### PR TITLE
[jit] Preserve module names in recursive script

### PR DESCRIPTION
**Pull request by driazati**
Turns
```
ScriptModule(
  (conv): ScriptModule()
  (lin): ScriptModule()
  (sub): ScriptModule()
)
```

into

```
ScriptModule(
  original=MyModule
  (conv): ScriptModule(original=Conv2d)
  (lin): ScriptModule(original=Linear)
  (sub): ScriptModule(original=Submodule)
)
```

Differential Revision: [D16862032](https://our.internmc.facebook.com/intern/diff/16862032/)

### CODE REVIEW

**Review by eellison — COMMENTED** [review]
Looks good but i think we should in some way indicate that they are script modules

@suo @jamesr66a @zdevito thoughts?

**Comment by driazati** [comment]
> Looks good but i think we should in some way indicate that they are script modules
> 
> @suo @jamesr66a @zdevito thoughts?

I changed it to use `extra_repr` instead which keeps `ScriptModule` around, see the updated output in the PR body above

**Comment by aazzolini** [comment]
I believe it's a bit confusing because I was under the impression that original was a member of the module (e.g. that i'd do isinstance(submodule.original, OriginalClass) to check types when in reality that is not needed .

File: torch/jit/__init__.py [hunk 1]
```diff
@@ -1589,6 +1589,11 @@ def __getstate__(self):
         def graph_for(self, *args, **kwargs):
             return self.forward.graph_for(*args, **kwargs)
 
+        def extra_repr(self):
```
Comment (suo) [inline torch/jit/__init__.py]: Can we also add an api to get this directly? orig_name or something?

**Review by suo — APPROVED** [review]
let's get this landed, someone was asking about it

**Pull request closed without merging**